File: AuctionServices/src/main/java/com/artauction/auctionservice/Entities/PostSold.java
package com.artauction.auctionservice.Entities;

import com.artauction.auctionservice.Config.IDs.PostSoldId;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "post_solds")
@Getter
@Setter
public class PostSold {
    // Attributes
    @EmbeddedId
    private PostSoldId id = new PostSoldId();

    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal finalPrice;
    private boolean isPaid = false;
}

File: Project/Backend/src/ArtAuction.Application/Entities/PostSold.cs
namespace ArtAuction.Application.Entities;

public class PostSold
{
    // Attributes
    public int BuyerId { get; set; }
    public int ArtworkPostId { get; set; } 
    public decimal FinalPrice { get; set; }
    public bool IsPaid { get; set; }

    // Relationships
    public Buyer Buyer { get; set; }
    public ArtworkPost ArtworkPost { get; set; }
}